nipoppy/sample_run_nipoppy.py

```diff
@@ -10,22 +10,56 @@
 import nipoppy.workflow.logger as my_logger
 # from nipoppy.workflow.tabular import generate_manifest
 from nipoppy.workflow.dicom_org import run_dicom_org
-from nipoppy.workflow.dicom_org import check_dicom_status
+from nipoppy.workflow import make_doughnut
 from nipoppy.workflow.bids_conv import run_bids_conv
 from nipoppy.workflow.proc_pipe.mriqc import run_mriqc
 from nipoppy.workflow.proc_pipe.fmriprep import run_fmriprep
 from nipoppy.workflow.catalog import get_new_proc_participants
 from nipoppy.workflow.catalog import generate_pybids_index
 from nipoppy.trackers import run_tracker
 
+# helper functions
+def get_proc_batches(proc_participants, MAX_BATCH, logger):
+    """ Generates MAX_BATCH participants to run at any given time before clean-up
+    """
+    n_proc_participants = len(proc_participants)
+    if n_proc_participants > MAX_BATCH: 
+        n_batches = int(np.ceil(n_proc_participants/MAX_BATCH))
+        logger.info(f"Running fmriprep in {n_batches} batches of at most {MAX_BATCH} participants each")
+        proc_participant_batches = np.array_split(proc_participants, n_batches)
+    else:
+        proc_participant_batches = [proc_participants]    
+
+    return proc_participant_batches
+
+def refresh_bids_db(global_configs, session_id, pipeline, ignore_patterns, logger):
+    """ Remove and rebuilds the bids_db for the given pipeline with the given ignore_patterns
+    """
+    DATASET_ROOT = global_configs["DATASET_ROOT"]
+    # remove old bids_db
+    sql_db_file = f"{DATASET_ROOT}/proc/bids_db_{pipeline}/bids_db.sqlite" 
+    logger.info(f"Removing old bids_db from {sql_db_file}")
+    if os.path.exists(sql_db_file):
+        os.remove(sql_db_file)
```

Comment:
Another way to do this would be to pass `reset_database=True` to the `BIDSLayout()` call in `generate_pybids_index()`. Not sure which one should be preferred